src/components/CalendarPopover/CalendarPopover.tsx

```diff
@@ -64,9 +70,45 @@ export class CalendarPopover extends React.Component<CalendarPopoverProps> {
       [POPOVER_DATA_KEYS.WithShadow]: withShadow,
       [POPOVER_DATA_KEYS.Animated]: animated,
       [POPOVER_DATA_KEYS.Shown]: isShown,
+      [POPOVER_DATA_KEYS.AutoFocused]: autoFocused,
     };
   };
 
+  _onClose = () => {
+    this.props.onClose();
+    this.props.autoFocused &&
+      this.lastActiveElement &&
+      this.lastActiveElement.focus();
```

Comment:
I think I misled you here.
This logic shouldn't be inside `_onClose` (I think I told you to put it here).
But actually this logic should be called on `componentWillUnmount` and when `prevProps.isShown && !this.props.isShown` in `componentDidUpdate`.